xcm/pallet-xcm/src/lib.rs

```diff
@@ -822,6 +913,88 @@ pub mod pallet {
 			Ok(())
 		}
 
+		fn do_reserve_withdraw_assets(
+			origin: OriginFor<T>,
+			dest: Box<VersionedMultiLocation>,
+			beneficiary: Box<VersionedMultiLocation>,
+			assets: Box<VersionedMultiAssets>,
+			fee_asset_item: u32,
+			maybe_weight_limit: Option<WeightLimit>,
+		) -> DispatchResult {
+			let origin_location = T::ExecuteXcmOrigin::ensure_origin(origin)?;
+			let dest = (*dest).try_into().map_err(|()| Error::<T>::BadVersion)?;
+			let beneficiary: MultiLocation = (*beneficiary)
+				.try_into()
+				.map_err(|()| Error::<T>::BadVersion)?;
+			let assets: MultiAssets = (*assets).try_into().map_err(|()| Error::<T>::BadVersion)?;
+
+			ensure!(
+				assets.len() <= MAX_ASSETS_FOR_TRANSFER,
+				Error::<T>::TooManyAssets
+			);
+			let value = (origin_location, assets.drain());
+			ensure!(
+				T::XcmReserveTransferFilter::contains(&value),
```

Comment:
I think that this is not a good filter, `T::XcmReserveTransferFilter` is used for checking assets for "reserve based transfer out", which means **"origin can do reserve(assets) here and send ReserveAssetDeposited(assets) to destination"**,

```
        /// Our XCM filter which messages to be reserve-transferred using the dedicated extrinsic must pass.
		type XcmReserveTransferFilter: Contains<(MultiLocation, Vec<MultiAsset>)>;
```

---

for opposite **"reserve withdraw assets"**, there should be another dedicated filter like `T::XcmReserveWithdrawFilter` which should check almost the same as `xcm_executor::Config::IsReserve`,


`IsReserve` - checks (assets, origin) if runtime trusts to **origin** that holds reserves for some **assets** (origin is here e.g. sibling parachain)
`T::XcmReserveWithdrawFilter` - should check (dest, assets), if runtime trusts to **dest** that holds reserves, that we want to withdraw